frontend/src/services/UserService.ts

```diff
@@ -109,7 +122,69 @@ export class UserService {
 
     }
 
+    public static async setCurrentUser (user: User | undefined) {
+
+        if ( user !== this._user ) {
+            LOG.info("setCurrentUser;", user)
+            this._user = user;
+            if ( this._observer.hasCallbacks(UserServiceEvent.CURRENT_USER_CHANGED) ) {
+                if (user){
+                    const email: string | undefined = EmailAuthSessionService.getEmailAddress();
+                    LOG.debug("User email;", email)
+
+                    const workspace = WorkspaceService.getCurrentWorkspaceId()
+                    if (email && workspace){
+                        const userList = await UserService.getWorkspaceUserList(workspace)
+
+                        if (userList  ) {
+
+                            const result = userList.filter(user => user.email === email)
+
+                            if (result ){
+                                LOG.info("Set user: ", result[0])
+                                this._user = result[0];
+                            } else {
+                                LOG.debug(`useSelectWorkspaceUserCallback: Did not find user to this email `, email);
+                            }
+
+                        }
+
+                    }
+
+                }
+
+                this._observer.triggerEvent(UserServiceEvent.CURRENT_USER_CHANGED);
+
+
+            }
+        }
+    }
+
     public static initialize () {
+        LOG.info(`Initializing user`);
+        this._initializeUser().catch((err) => {
+            LOG.error(`ERROR: Could not initialize user: `, err);
+        });
+    }
+
+    public static async _initializeUser () {
+        const email: string | undefined = EmailAuthSessionService.getEmailAddress();
+        LOG.debug("User email;", email)
+        const workspace = WorkspaceService.getCurrentWorkspaceId()
+        if (email && workspace) {
+            const userList = await UserService.getWorkspaceUserList(workspace)
```

Comment:
Current users should not use the workspace user list to fetch records... We should have a profile end point to fetch current user information.

Also this logic should probably not be in the UserService at all. ProfileService would be better.